### PR TITLE
feat(clerk-js): Build ui of <OrganizationList/>

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

This PR implements the UI of OrganizationList (see images below).
In a follow-up PR we will expose the component to our packages

## Memberships (joined orgs)
image
## Memberships (joined orgs) with personal account
image

## Suggestion pending approval
image
## Suggestion request to join
image
## Invitation join
image

## Invitation + memberships
image

### CODE REVIEW

**Comment by panteliselef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @panteliselef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.28.1-snapshot.d218edb |
| @clerk/chrome-extension | 0.4.0-snapshot.d218edb |
| @clerk/clerk-js | 4.57.0-snapshot.d218edb |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.20-snapshot.d218edb |
| @clerk/fastify | 0.6.5-snapshot.d218edb |
| gatsby-plugin-clerk | 4.4.6-snapshot.d218edb |
| @clerk/localizations | 1.26.0-snapshot.d218edb |
| @clerk/nextjs | 4.24.0-snapshot.d218edb |
| @clerk/clerk-react | 4.25.0-snapshot.d218edb |
| @clerk/remix | 2.10.1-snapshot.d218edb |
| @clerk/clerk-sdk-node | 4.12.4-snapshot.d218edb |
| @clerk/shared | 0.22.0 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.51.0-snapshot.d218edb |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.28.1-snapshot.d218edb
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.4.0-snapshot.d218edb
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.57.0-snapshot.d218edb
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.20-snapshot.d218edb
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.6.5-snapshot.d218edb
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.4.6-snapshot.d218edb
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.26.0-snapshot.d218edb
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.24.0-snapshot.d218edb
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.25.0-snapshot.d218edb
```
```sh
# @clerk/remix
npm i @clerk/remix@2.10.1-snapshot.d218edb
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.12.4-snapshot.d218edb
```
```sh
# @clerk/shared
npm i @clerk/shared@0.22.0
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.51.0-snapshot.d218edb
```

**Comment by panteliselef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @panteliselef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.28.1-snapshot.890c1b9 |
| @clerk/chrome-extension | 0.4.0-snapshot.890c1b9 |
| @clerk/clerk-js | 4.57.0-snapshot.890c1b9 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.20-snapshot.890c1b9 |
| @clerk/fastify | 0.6.5-snapshot.890c1b9 |
| gatsby-plugin-clerk | 4.4.6-snapshot.890c1b9 |
| @clerk/localizations | 1.26.0-snapshot.890c1b9 |
| @clerk/nextjs | 4.24.0-snapshot.890c1b9 |
| @clerk/clerk-react | 4.25.0-snapshot.890c1b9 |
| @clerk/remix | 2.10.1-snapshot.890c1b9 |
| @clerk/clerk-sdk-node | 4.12.4-snapshot.890c1b9 |
| @clerk/shared | 0.22.0 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.51.0-snapshot.890c1b9 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.28.1-snapshot.890c1b9
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.4.0-snapshot.890c1b9
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.57.0-snapshot.890c1b9
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.20-snapshot.890c1b9
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.6.5-snapshot.890c1b9
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.4.6-snapshot.890c1b9
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.26.0-snapshot.890c1b9
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.24.0-snapshot.890c1b9
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.25.0-snapshot.890c1b9
```
```sh
# @clerk/remix
npm i @clerk/remix@2.10.1-snapshot.890c1b9
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.12.4-snapshot.890c1b9
```
```sh
# @clerk/shared
npm i @clerk/shared@0.22.0
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.51.0-snapshot.890c1b9
```

**Comment by panteliselef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @panteliselef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.28.1-snapshot.58eb7b1 |
| @clerk/chrome-extension | 0.4.0-snapshot.58eb7b1 |
| @clerk/clerk-js | 4.57.0-snapshot.58eb7b1 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.20-snapshot.58eb7b1 |
| @clerk/fastify | 0.6.5-snapshot.58eb7b1 |
| gatsby-plugin-clerk | 4.4.6-snapshot.58eb7b1 |
| @clerk/localizations | 1.26.0-snapshot.58eb7b1 |
| @clerk/nextjs | 4.24.0-snapshot.58eb7b1 |
| @clerk/clerk-react | 4.25.0-snapshot.58eb7b1 |
| @clerk/remix | 2.10.1-snapshot.58eb7b1 |
| @clerk/clerk-sdk-node | 4.12.4-snapshot.58eb7b1 |
| @clerk/shared | 0.22.0 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.51.0-snapshot.58eb7b1 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.28.1-snapshot.58eb7b1
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.4.0-snapshot.58eb7b1
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.57.0-snapshot.58eb7b1
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.20-snapshot.58eb7b1
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.6.5-snapshot.58eb7b1
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.4.6-snapshot.58eb7b1
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.26.0-snapshot.58eb7b1
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.24.0-snapshot.58eb7b1
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.25.0-snapshot.58eb7b1
```
```sh
# @clerk/remix
npm i @clerk/remix@2.10.1-snapshot.58eb7b1
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.12.4-snapshot.58eb7b1
```
```sh
# @clerk/shared
npm i @clerk/shared@0.22.0
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.51.0-snapshot.58eb7b1
```

**Comment by panteliselef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @panteliselef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.28.1-snapshot.454688d |
| @clerk/chrome-extension | 0.4.0-snapshot.454688d |
| @clerk/clerk-js | 4.57.0-snapshot.454688d |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.20-snapshot.454688d |
| @clerk/fastify | 0.6.5-snapshot.454688d |
| gatsby-plugin-clerk | 4.4.6-snapshot.454688d |
| @clerk/localizations | 1.26.0-snapshot.454688d |
| @clerk/nextjs | 4.24.0-snapshot.454688d |
| @clerk/clerk-react | 4.25.0-snapshot.454688d |
| @clerk/remix | 2.10.1-snapshot.454688d |
| @clerk/clerk-sdk-node | 4.12.4-snapshot.454688d |
| @clerk/shared | 0.22.0 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.51.0-snapshot.454688d |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.28.1-snapshot.454688d
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.4.0-snapshot.454688d
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.57.0-snapshot.454688d
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.20-snapshot.454688d
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.6.5-snapshot.454688d
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.4.6-snapshot.454688d
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.26.0-snapshot.454688d
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.24.0-snapshot.454688d
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.25.0-snapshot.454688d
```
```sh
# @clerk/remix
npm i @clerk/remix@2.10.1-snapshot.454688d
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.12.4-snapshot.454688d
```
```sh
# @clerk/shared
npm i @clerk/shared@0.22.0
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.51.0-snapshot.454688d
```

**Comment by panteliselef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @panteliselef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.28.1-snapshot.15b0738 |
| @clerk/chrome-extension | 0.4.0-snapshot.15b0738 |
| @clerk/clerk-js | 4.57.0-snapshot.15b0738 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.20-snapshot.15b0738 |
| @clerk/fastify | 0.6.5-snapshot.15b0738 |
| gatsby-plugin-clerk | 4.4.6-snapshot.15b0738 |
| @clerk/localizations | 1.26.0-snapshot.15b0738 |
| @clerk/nextjs | 4.24.0-snapshot.15b0738 |
| @clerk/clerk-react | 4.25.0-snapshot.15b0738 |
| @clerk/remix | 2.10.1-snapshot.15b0738 |
| @clerk/clerk-sdk-node | 4.12.4-snapshot.15b0738 |
| @clerk/shared | 0.22.0 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.51.0-snapshot.15b0738 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.28.1-snapshot.15b0738
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.4.0-snapshot.15b0738
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.57.0-snapshot.15b0738
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.20-snapshot.15b0738
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.6.5-snapshot.15b0738
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.4.6-snapshot.15b0738
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.26.0-snapshot.15b0738
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.24.0-snapshot.15b0738
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.25.0-snapshot.15b0738
```
```sh
# @clerk/remix
npm i @clerk/remix@2.10.1-snapshot.15b0738
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.12.4-snapshot.15b0738
```
```sh
# @clerk/shared
npm i @clerk/shared@0.22.0
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.51.0-snapshot.15b0738
```

**Comment by panteliselef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @panteliselef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.28.1-snapshot.1e6df0c |
| @clerk/chrome-extension | 0.4.0-snapshot.1e6df0c |
| @clerk/clerk-js | 4.57.0-snapshot.1e6df0c |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.20-snapshot.1e6df0c |
| @clerk/fastify | 0.6.5-snapshot.1e6df0c |
| gatsby-plugin-clerk | 4.4.6-snapshot.1e6df0c |
| @clerk/localizations | 1.26.0-snapshot.1e6df0c |
| @clerk/nextjs | 4.24.0-snapshot.1e6df0c |
| @clerk/clerk-react | 4.25.0-snapshot.1e6df0c |
| @clerk/remix | 2.10.1-snapshot.1e6df0c |
| @clerk/clerk-sdk-node | 4.12.4-snapshot.1e6df0c |
| @clerk/shared | 0.22.0 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.51.0-snapshot.1e6df0c |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.28.1-snapshot.1e6df0c
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.4.0-snapshot.1e6df0c
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.57.0-snapshot.1e6df0c
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.20-snapshot.1e6df0c
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.6.5-snapshot.1e6df0c
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.4.6-snapshot.1e6df0c
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.26.0-snapshot.1e6df0c
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.24.0-snapshot.1e6df0c
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.25.0-snapshot.1e6df0c
```
```sh
# @clerk/remix
npm i @clerk/remix@2.10.1-snapshot.1e6df0c
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.12.4-snapshot.1e6df0c
```
```sh
# @clerk/shared
npm i @clerk/shared@0.22.0
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.51.0-snapshot.1e6df0c
```

File: packages/clerk-js/src/ui/components/OrganizationList/OrganizationListPage.tsx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+import { useInView } from '../../hooks';
+import { CreateOrganizationForm } from '../CreateOrganization/CreateOrganizationForm';
+
+import { PreviewListItems, PreviewListSpinner } from './shared';
+import { InvitationPreview } from './UserInvitationList';
+import { MembershipPreview, PersonalAccountPreview } from './UserMembershipList';
+import { SuggestionPreview } from './UserSuggestionList';
+import { organizationListParams } from './utils';
+
+const useFetchList = () => {
```
Comment (anagstef) [inline packages/clerk-js/src/ui/components/OrganizationList/OrganizationListPage.tsx:15]: 🙃  Nitpicking comment: 

Can we rename this one to something more relevant? Like, `useCoreOrganizationListWithRef` or `useCoreOrganizationListInView` or sth else. `useFetchList` sounds a little generic.

File: packages/clerk-js/src/ui/components/OrganizationList/utils.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import type { useCoreOrganizationList } from '../../contexts';
+
+export const organizationListParams = {
```
Comment (anagstef) [inline packages/clerk-js/src/ui/components/OrganizationList/utils.ts:3]: ❓ What's the use of these hardcoded params here? 🤔

**Review by anagstef — APPROVED** [review]
👍

File: packages/clerk-js/src/ui/components/OrganizationList/OrganizationListPage.tsx (hunk 1 above)
Comment (panteliselef) [reply packages/clerk-js/src/ui/components/OrganizationList/OrganizationListPage.tsx:15]: Makes sense

File: packages/clerk-js/src/ui/components/OrganizationList/utils.ts (hunk 2 above)
Comment (panteliselef) [reply packages/clerk-js/src/ui/components/OrganizationList/utils.ts:3]: So swr fetches data and cached them based on a key.

As key we use these params. For example 
`useOrganizatioList({userMemership:true})` will store data in a different place than `useOrganizationList({userMemership:{infinite:true}})`.

Inside OrganizationList we call useOrganizationList many times and for consistency I've added this obj as a source of truth for all of them.

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**